controllers/humiocluster_tls.go

```diff
@@ -199,3 +207,63 @@ func constructNodeCertificate(hc *humiov1alpha1.HumioCluster, nodeSuffix string)
 		},
 	}
 }
+
+func (r *HumioClusterReconciler) waitForNewNodeCertificate(ctx context.Context, hc *humiov1alpha1.HumioCluster, expectedCertCount int) error {
+	for i := 0; i < waitForNodeCertificateTimeoutSeconds; i++ {
+		existingNodeCertCount, err := r.nodeCertificateCount(ctx, hc)
+		if err != nil {
+			return err
+		}
+		r.Log.Info(fmt.Sprintf("validating new pod certificate was created. expected pod certificate count %d, current pod certificate count %d", expectedCertCount, existingNodeCertCount))
+		if existingNodeCertCount >= expectedCertCount {
+			return nil
+		}
+		time.Sleep(time.Second * 1)
+	}
+	return fmt.Errorf("timed out waiting to validate new pod certificate was created")
+}
+
+func (r *HumioClusterReconciler) nodeCertificateCount(ctx context.Context, hc *humiov1alpha1.HumioCluster) (int, error) {
+	certificates, err := kubernetes.ListCertificates(r, hc.Namespace, kubernetes.MatchingLabelsForHumio(hc.Name))
+	if err != nil {
+		return -1, err
+	}
+
+	existingNodeCertCount := 0
+	for _, cert := range certificates {
+		if strings.HasPrefix(cert.Name, fmt.Sprintf("%s-core", hc.Name)) {
+			existingNodeCertCount++
+
+			// Check if we should update the existing certificate
+			certForHash := constructNodeCertificate(hc, "")
+
+			// Keystores will always contain a new pointer when constructing a certificate.
+			// To work around this, we override it to nil before calculating the hash,
+			// if we do not do this, the hash will always be different.
+			certForHash.Spec.Keystores = nil
+
+			desiredCertificateHash := helpers.AsSHA256(certForHash)
```

Comment:
Perhaps we could do the same json marshal here before calculating the hash?

Reply:
Yes.